usr/lib/common/new_host.c

```diff
@@ -3939,6 +3939,9 @@ CK_RV SC_GenerateRandom(STDLL_TokData_t *tokdata, ST_SESSION_HANDLE *sSession,
         goto done;
     }
 
+    if (ulRandomLen == 0)
+        goto done;
+
```

Comment:
Yes, makes sense. Unfortunately the pkcs#11 spec does not say what to do if the number of bytes is (perhaps accidentally) zero. 
Variants would be: zero out the buffer before returning? Or return with an error, e.g. CKR_ARGUMENTS_BAD, telling the application that the buffer shall not be used after the call to C_SeedRandom?

Reply:
> Variants would be: zero out the buffer before returning?

Hmm, zeroing out a buffer of 0 bytes in length :-) 

> Or return with an error, e.g. CKR_ARGUMENTS_BAD

I thought about that, but is it really an error? The application asked for 0 bytes of random data and got 0 bytes of random data. So no error in my opinion.

Actually, this fix is because Coverty complained about a possible NULL pointer access when pRandom = NULL. 
The code already checks for `if (!pRandomData && ulRandomLen != 0)` and returns an error if so.
So pRandom is only allowed to be NULL if ulRandomLen is zero. This should not have caused any real NULL pointer access in the further code anyway, but shortcutting this makes sense, I would say.